### PR TITLE
Payment Types ticket 3

### PR DESCRIPTION
**Pull request by LanceP51**
## Proposed changes
Ticket 3 completes all endpoints to GET, GET single, POST, PUT, DELETE to the PaymentType table/list.

## Related Issues
This ticket will function independently from other tickets once you have run the SQL script to build the database and seed data using the file "bangazonapi-team-horrohawk/TablesAndSeedData" from our code repository.

## Steps To Test
clone down the project to your computer
cd into BangazonAPI
from command line, "start BangazonAPI.sln" to open in Visual Studio
create the database and seed the data by running a quick new query with the data from our file on the Git repo at bangazonapi-team-horrohawk/TablesAndSeedData
Launch the Build from Visual Studio to view in your browser
be sure you are in the route "paymenttype" and you should see a list of PaymentTypes
Add "/1" to view  a single payment type
Open your Postman desktop application and insert url from browser, showing the "/paymenttype" route
Use the POST method to insert
{
        "accountNumber": 900356875,
        "name": "Target Visa",
        "customerId": 3
}
See this new object appear in the database
Change to the PUT method, add /{number of new Id here}
Change object to
{
        "accountNumber": 900356875,
        "name": "Target Discover",
        "customerId": 3
}
GET this route again and see the new change
DELETE from this route with "/1" and you will see a "Cannot Delete This One"
Try to DELETE from a route that you created, such as "/4" and you will be able to delete this because there is no associated foreign key from the Order Table

Please Respond with any issues.

## What kind of change does this PR introduce?
Please check one of the following:
- [ ] Bug fix
- [ ] Documentation update
- [x] New feature
- [ ] Other

### CODE REVIEW

**Comment by Jengwen** [comment]
All features working, change PR to have steps to test under the heading steps to test.

**Comment by jordan-castelloe** [comment]
Testing!

File: BangazonAPI/Controllers/PaymentTypeController.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"INSERT INTO PaymentType (Name, AcctNumber, CustomerId)
+                                        OUTPUT INSERTED.Id
+                                        VALUES (@Name, @AcctNumber, @CustomerId)";
+                    cmd.Parameters.Add(new SqlParameter("@Name", currentPayment.name));
+                    cmd.Parameters.Add(new SqlParameter("@AcctNumber", currentPayment.accountNumber));
```
Comment (jordan-castelloe) [inline BangazonAPI/Controllers/PaymentTypeController.cs]: When I try to `POST` or `PUT` with the dummy data you gave me in your PR, it inserts a new payment type into the database with an account number of 0.  Your dummy data contains an account number, but it looks like it's not making it to the database.

**Review by jordan-castelloe — CHANGES_REQUESTED** [review]

File: BangazonAPI/Controllers/PaymentTypeController.cs (hunk 1 above)
Comment (LanceP51) [reply BangazonAPI/Controllers/PaymentTypeController.cs]: that was simple instructions mishap. fixed the instructions so everything should work fine now!

Comment (jordan-castelloe) [reply BangazonAPI/Controllers/PaymentTypeController.cs]: Okay, thanks!

**Review by jordan-castelloe — APPROVED** [review]
You're good! Make sure you run the tests again after you resolve these merge conflicts.

**Pull request merged**